src/Catalyst.Core.Modules.Dfs/CoreApi/DnsApi.cs

```diff
@@ -0,0 +1,162 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs.CoreApi;
+using Lib.P2P;
+using Makaretu.Dns;
+
+namespace Catalyst.Core.Modules.Dfs.CoreApi
+{
+    internal sealed class DnsApi : IDnsApi
+    {
+        private readonly IDnsClient _dnsClient;
+
+        public DnsApi(DotClient dnsClient)
+        {
+            _dnsClient = dnsClient;
+        }
+
+        public async Task<string> ResolveNameAsync(string name,
+            bool recursive = false,
+            bool nocache = false,
+            CancellationToken cancel = default)
+        {
+            do
+            {
+                if (name.StartsWith("/ipns/"))
+                {
+                    name = name.Substring(6);
+                }
+
+                var parts = name.Split('/').Where(p => p.Length > 0).ToArray();
+                if (parts.Length == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)